src/uri/uri_parser.cr

```diff
@@ -0,0 +1,245 @@
+# :nodoc:
+class URIParser
```

Comment:
Why not `URI::Parser` instead? (that way it does not clobber the global namespace)